### DISCUSSION TITLE
*** Using Compiler 'V5.06 update 7 (build 960)', folder: 'D:\keil5 mdk\core\ARM\ARMCC\Bin' Build target 'stm32f103xb_if' Before Build - User command #1: ..\..\..\tools\pre_build_script.bat Python 3.11.5  f:\1\DAPLink-main\projectfiles\uvision\stm32f103xb_if>python ..\..\..\tools\pre_build_script.py  fatal: Not a valid object name HEAD #> Getting git description #> ERROR: Failed to get git description, do you have git.exe in your PATH environment variable? ".\build\stm32f103xb_if.axf" - 1 Error(s), 0 Warning(s). Target not created. Build Time Elapsed:  00:00:01

### DISCUSSION THREAD

**Discussion by LONG300002** [ARMmbed/DAPLink#1054] [Q&A]
*** Using Compiler 'V5.06 update 7 (build 960)', folder: 'D:\keil5 mdk\core\ARM\ARMCC\Bin'
Build target 'stm32f103xb_if'
Before Build - User command #1: ..\..\..\tools\pre_build_script.bat
Python 3.11.5

f:\1\DAPLink-main\projectfiles\uvision\stm32f103xb_if>python ..\..\..\tools\pre_build_script.py 
fatal: Not a valid object name HEAD
#> Getting git description
#> ERROR: Failed to get git description, do you have git.exe in your PATH environment variable?
".\build\stm32f103xb_if.axf" - 1 Error(s), 0 Warning(s).
Target not created.
Build Time Elapsed:  00:00:01

This problem has troubled me for a long time, git and python have correctly added the path

**Comment by mathias-arm** [comment]
See https://github.com/ARMmbed/DAPLink/issues/645#issuecomment-1752943533
